lib/Serialization/SILFormat.h

```diff
@@ -312,6 +356,7 @@ namespace sil_block {
                      BCFixed<1>,  // is distributed
                      BCFixed<1>,  // is runtime accessible
                      BCFixed<1>,  // are lexical lifetimes force-enabled
+                     BCFixed<1>,  // zombie
```

Comment:
I would not serialize this flag. Instead just serialize zombie functions as regular functions.

However, if you need this flag to disable linkage verification of such functions, it's better to call it something like "onlyReferencedByDebugInfo".

Reply:
Sometimes, functions that I serialize just for debug scopes get deserialized by the compiler in a non debug scopes context, e.g. by the generic specializer. In those cases, I try to resurrect the function which causes verification failures because the compiler is expecting the function to have a body which it does not. I guess i can get around that by just not resurrecting declarations that break these invariants, e.g. shared functions that are declarations only etc.. I will revisit this once I have made more progress on the PR

Reply:
I think what you should do is to _not_ try to de-serialize a "onlyReferencedByDebugInfo" function if it's created from an optimization pass, like the specializer. Instead just create a new function in this case.
Only when de-serializing debug info and a function is referenced from debug info and it doesn't exist, yet (either as a "real" function or as a zombie function), then de-serialize a "onlyReferencedByDebugInfo" function.

Reply:
Yes that is what I am doing right now here. I just meant to say that maybe I can do this without storing the extra onlyReferencedByDebugInfo bit https://github.com/swiftlang/swift/blob/07401c0b474eeeaec4c9159f4cfd0d015d7f1fb6/lib/Serialization/DeserializeSIL.cpp#L780

Reply:
I just return null in which case I think the optimizer handles creation of a new function